rest_polymorphic/serializers.py

```diff
@@ -120,7 +122,13 @@ def _get_serializer_from_model_or_instance(self, model_or_instance):
 
         for klass in model.mro():
             if klass in self.model_serializer_mapping:
-                return self.model_serializer_mapping[klass]
+                serializer = self.model_serializer_mapping[klass]
+                if isinstance(serializer, str):
+                    serializer = import_string(serializer)(
```

Comment:
``django.utils.module_loading.import_string`` is a convenience wrapper around ``importlib.import_module``. effects are identical and flex-fields basically reimplements import_string in `_import_serializer_class`.

Unfortunatelly, I was not able to find tests in flex-fields that actually use that feature. I suspect it might break on situations like the example below.

The thing is it *can* work in ``__init__`` for simpler cases but it breaks in the really interesting ones. Consider this example: 


	class ZSerializer(PolymorhicSerializer)
		model_serializer_mapping = {
			ModelX: 'XSerializer',
			ModelY: 'YSerializer',
		}

	class XSerializer(Serializer):
		# some fields

	class YSerializer(Serializer):
		# other fields
		z = ZSerializer()


Once you start using a nested field like ``z = ZSerializer()``, the ``__init__`` approach breaks down. You must init the ``ZSerializer`` with ``()`` there. That means ``ZSerializer`` inits inside the definition of ``YSerializer``. When it inits it tries to load the string for ``YSerializer``, but loading ``YSerializer`` cannot succeed as we just went into a cyclic import.

Basically the culprit is the ``ZSerializer()``. As this is a DRF idiom there is no way around it and as a result "transitive self-referential" imports cannot happen in serializer constructors.

Also, rearranging the classes into separate files will not solve the issue.

Not sure if that is something that can happen with how flex-fields are supposed to be used. I have never used it.


Is that logic comprehensible or did I miss something here?

Reply:
@apirobot i know you are pretty pressed for time, but did you get a chance to think about this? thanks!

Reply:
@apirobot thank you for taking the time! my test only covers basic functionality and not the feature i outlined in the comment above. i'll construct a test for that as well and test both versions.